ACKategories/Classes/Control+Blocks.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  Control.swift
+//  Pods
+//
+//  Created by Jan Mísař on 31.08.16.
+//
+//
+
+import UIKit
+
+class CollectionWrapper<T: UIControl> {
+    var targets: [UIControlEvents.RawValue: ActionWrapper<T>] = [:]
+}
+
+class ActionWrapper<T: UIControl> {
+    var action: (T -> Void)
+
+    init(action: (T -> Void)) {
+        self.action = action
+    }
+
+    @objc func invokeAction(sender: UIControl) {
+        if let sender = sender as? T {
+            action(sender)
+        }
+    }
+}
+
+public protocol UIControlEventHandling {
+    func on(events: UIControlEvents, handler: Self -> Void)
+}
+
+private var actionKey: UInt8 = 0
+
+public extension UIControlEventHandling where Self: UIControl {
+
+    /**
+     Register action block to be executed on defined events.
+
+     - parameter events Events to fire action block
+     - parameter handler Actionblock to be executed
+     */
+    func on(events: UIControlEvents, handler: Self -> Void) {
+
+        var targetsWrapper: CollectionWrapper<Self>! = nil
```

Comment:
I think this can be let, if it gets initialized before its used, the compiler will understand it.